CKSIncrementalStore/CKSIncrementalStoreSyncOperation.swift

```diff
@@ -639,7 +639,7 @@ class CKSIncrementalStoreSyncOperation: NSOperation {
         if self.localStoreMOC!.hasChanges
         {
             var error:NSErrorPointer = nil
-            self.localStoreMOC!.save(error)
+            self.localStoreMOC!.save(error) // TODO: saving one object at a time (or recovering and saving as much as possible) would be more robust (and catching exceptions during all the above). Otherwise a failure will result in the entire change set being lost. Also, it probably should revert/rollback on failure also.
```

Comment:
This can leave the local database in an inconsistent state. Changes to the local store and CloudKit servers should be atomic i.e All or none to maintain consistency.